IntegrationTests/Tests/IntegrationTests/SwiftPMTests.swift

```diff
@@ -51,7 +51,7 @@ final class SwiftPMTests: XCTestCase {
 
     func testSwiftBuild() throws {
         #if os(Linux)
```

Comment:
suggestion: can we add a test where we try to instantiate an executable wth the name `main` ?  

something like 
```
swift package --package-path main init --type executable
swift run  or
```

Reply:
Good idea, I've updated the PR to handle this edge case and added a new test.